### PR TITLE
Implement `Selector` (#574 part4)

### PR DESCRIPTION
**Pull request by inikulin**
In this part:
- Use selector machinery errors for action selectors on construction
- Test selector from function
- Throw if selector returns anything except null, undefined or DOM node

\cc @VasilyStrelyaev @helen-dikareva @AlexanderMoskovkin

### CODE REVIEW

File: test/functional/fixtures/api/es-next/drag/test.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('[API] Drag actions', function () {
         it('Should validate selector', function () {
             return runTests('./testcafe-fixtures/drag-test.js', 'Drag to offset with incorrect selector', { shouldFail: true, only: 'chrome' })
                 .catch(function (errs) {
-                    expect(errs[0]).to.contain('The selector is expected to be a string, but it was object.');
+                    expect(errs[0]).to.contain('Action selector error:  Selector code is expected to be specified as a function or string, but "object" was passed.');
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/drag/test.js]: Can't find where this message is built in code, so I'll put it here...
What do you mean by saying that selector code is expected to be specified as a string? CSS selectors? But that's not code. I would think of "code specified as a string" as a string with js code.

What's the template this string is built from?

**Comment by VasilyStrelyaev** [comment]
\r-

File: test/functional/fixtures/api/es-next/drag/test.js (hunk 1 above)
Comment (inikulin) [inline test/functional/fixtures/api/es-next/drag/test.js]: https://github.com/DevExpress/testcafe/blob/master/src/errors/runtime/message.js#L26

**Comment by AlexanderMoskovkin** [comment]
lgtm but rebase required

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: src/errors/test-run/index.js [hunk 2]
```diff
@@ -70,14 +70,24 @@ export class DomNodeClientFunctionResultError extends TestRunErrorBase {
     }
 }
 
+// Selector errors
```
Comment (helen-dikareva) [inline src/errors/test-run/index.js]: Why we need to two groups of `Selector errors`? https://github.com/DevExpress/testcafe/pull/603/files#diff-ab5af4bc525a2e3c00245caa5503456eR219

Comment (inikulin) [inline src/errors/test-run/index.js]: One generic and one for Actions

Comment (helen-dikareva) [inline src/errors/test-run/index.js]: OK

**Comment by helen-dikareva** [comment]
lgtm

**Comment by inikulin** [comment]
ping @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**